File: packages/bouncer-py/examples/basic_claim.py
from __future__ import annotations

import bouncer


def main() -> None:
    db = bouncer.open("app.sqlite3")
    db.bootstrap()

    result = db.claim("scheduler", "worker-a", ttl_ms=30_000)
    if result.acquired:
        print(f"acquired token {result.lease.token}")
    else:
        print(f"busy: currently owned by {result.current.owner}")


if __name__ == "__main__":
    main()

File: packages/bouncer/examples/basic_claim.rs
use std::time::Duration;

use bouncer::{Bouncer, ClaimResult};

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let db = Bouncer::open("app.sqlite3")?;
    db.bootstrap()?;

    match db.claim("scheduler", "worker-a", Duration::from_secs(30))? {
        ClaimResult::Acquired(lease) => {
            println!("acquired {} with token {}", lease.name, lease.token);
        }
        ClaimResult::Busy(current) => {
            println!("busy: currently owned by {}", current.owner);
        }
    }

    Ok(())
}
